### PR TITLE
[release byod] fix ml requirements file selection

### PR DESCRIPTION
**Pull request by aslonnie**
and also change ecr to registry, as it might be google's too.

### CODE REVIEW

**Comment by aslonnie** [comment]
one more place that I found about picking ml-or-not that is based on the old "cpu-or-not" logic.

File: release/ray_release/test.py [hunk 1]
```diff
@@ -251,15 +251,15 @@ def get_byod_image_tag(self) -> str:
         }
         return f"{self.get_byod_base_image_tag()}-{dict_hash(custom_info)}"
 
-    def _use_byod_ml_image(self) -> bool:
+    def use_byod_ml_image(self) -> bool:
```
Comment (aslonnie) [inline release/ray_release/test.py:254]: maybe if using ml image should have it's own field? like technically speaking, we also have `ray-ml:cpu` image, just never used in release tests..

**Comment by aslonnie** [comment]
this probably only affects @xwjiang2010 's new test.

File: release/ray_release/byod/build.py [hunk 2]
```diff
@@ -124,11 +124,12 @@ def build_anyscale_base_byod_images(tests: List[Test]) -> None:
         and int(time.time()) - start < BASE_IMAGE_WAIT_TIMEOUT
     ):
         for byod_image, test in to_be_built.items():
-            byod_requirements = (
-                f"{REQUIREMENTS_BYOD}_{test.get('python', PYTHON_VERSION)}.txt"
-                if test.get_byod_type() == "cpu"
-                else f"{REQUIREMENTS_ML_BYOD}_{test.get('python', PYTHON_VERSION)}.txt"
-            )
+            py_version = test.get("python", PYTHON_VERSION)
```
Comment (can-anyscale) [inline release/ray_release/byod/build.py:127]: nit: test.get_python_version()

**Review by can-anyscale — APPROVED** [review]

File: release/ray_release/byod/build.py (hunk 2 above)
Comment (aslonnie) [reply release/ray_release/byod/build.py]: done.. why were you not using that in the past?

**Pull request merged**